### PR TITLE
Slate improvements

### PR DESCRIPTION
**Pull request by HaukurPall**
- Updated all dependencies expect react-scripts which needs to be held back until craco, semantic-ui-less and friends update.
- Using updated React Router API.
- Presisting redux state to local storage so that language preferences, source and target languages (and more) are stored between sessions.
- Fixed the Slate placeholder location for Firefox.
- We no longer specify the translation model to use for translations, we let the backend decide for us based on the languages selected.
- Added SlateJS history (undo - redo)
- Reimplemented the SlateJS editors.
- Made the translation side of the editor readOnly and added a "copy text" button to copy the contents to a clipboard.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -27,18 +28,40 @@ Sentry.init({
...
+      </Suspense>
+    </PersistGate>
+  </Provider>
 );
 
 // If you want your app to work offline and load faster, you can change
 // unregister() to register() below. Note this comes with some pitfalls.
 // Learn more about service workers: https://bit.ly/CRA-PWA
-serviceWorker.unregister();
+// serviceWorker.unregister();
```
Comment (vesteinn) [inline src/index.js:67]: Comment out or remove? Should this be `register`?

File: src/components/SlateEditor/index.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    //   }),
+    // ];
+    // navigator.clipboard.write(data).then(
+    //   function () {
+    //     console.log("Copied to clipboard successfully!");
+    //   },
+    //   function () {
+    //     console.error("Unable to write to clipboard. :-(");
+    //   }
+    // );
```
Comment (vesteinn) [inline src/components/SlateEditor/index.js:205]: Remove?

**Review by vesteinn — APPROVED** [review]

File: src/components/SlateEditor/index.js (hunk 2 above)
Comment (HaukurPall) [reply src/components/SlateEditor/index.js:205]: Removed with a patch

File: src/index.js (hunk 1 above)
Comment (HaukurPall) [reply src/index.js:67]: There is no support for a serviceWorker. Commenting out this line did not introduce a serviceWorker

**Pull request merged**